### ISSUE TITLE
Should examples go in examples/beman/exemplar instead of examples?

### ISSUE THREAD

**Issue by camio** [bemanproject/exemplar#86]
This would make the directory structure consistent with what's done in `include/`, `tests/`, and `src/`. One might consider it a drawback that examples, which are very important to new users, are buried so deep, but the GitHub interface makes it one click either way.

One benefit, aside from consistency, is that it becomes easier to "merge" repositories in case we ever wanted to provide an "all in one" distribution.

**Comment by ClausKlein** [comment]
I do not unterstand the idea behind this deep structure?

You try to makes name unique without any need?
If you plan to build a `super project` for all beman libraries, that is the wrong way.

Do you know your build tree?
```bash
bash-5.2$ tree -d -L5
.
|-- build
|   `-- clang-debug
|       |-- CMakeFiles
|       |   |-- 3.31.1
|       |   |   |-- CompilerIdC
|       |   |   `-- CompilerIdCXX
|       |   |-- CMakeScratch
|       |   `-- pkgRedirects
|       |-- _deps
|       |   |-- googletest-build
|       |   |   |-- CMakeFiles
|       |   |   |-- googlemock
|       |   |   `-- googletest
|       |   |-- googletest-src
|       |   |   |-- ci
|       |   |   |-- docs
|       |   |   |-- googlemock
|       |   |   `-- googletest
|       |   `-- googletest-subbuild
|       |       |-- CMakeFiles
|       |       `-- googletest-populate-prefix
|       |-- bin
|       |-- examples
|       |   `-- CMakeFiles
|       |       |-- beman.exemplar.examples.identity_as_default_projection.dir
|       |       `-- beman.exemplar.examples.identity_direct_usage.dir
|       |-- lib
|       |-- src
|       |   `-- beman
|       |       `-- exemplar
|       `-- tests
|           `-- beman
|               `-- exemplar
|-- examples
|-- include
|   `-- beman
|       `-- exemplar
|-- src
|   `-- beman
|       `-- exemplar
`-- tests
    `-- beman
        `-- exemplar

44 directories
```

And the installation tree:
```
bash-5.2$ tree /tmp/beman/usr/local/
/tmp/beman/usr/local/
|-- include
|   `-- beman
|       `-- exemplar
|           `-- identity.hpp
`-- lib
    `-- libbeman.exemplar.a

5 directories, 2 files
```

**Comment by neatudarius** [comment]
> One benefit, aside from consistency, is that it becomes easier to "merge" repositories in case we ever wanted to provide an "all in one" distribution.

@camio,  I agree with both reasons. Consistency is simple, if we claim it right now when Beman org is small, we can have it. If we try do it later, it may be hard.


> I do not unterstand the idea behind this deep structure?
> You try to makes name unique without any need?
@ClausKlein , as you said, we intentionally want that structure which helps us if working or installing multiple Beman libraries in same root directory. Maybe it's not a need, but a choice to have a simple organization.
<img width="821" alt="image" src="https://github.com/user-attachments/assets/9debf814-5676-40f1-b4e6-88f70707d3a8">
Wanted and expected users to use `#include <beman/$short_name/header.hpp>`. I think some of the discussions on this topic were on this PR: https://github.com/bemanproject/beman/pull/18/files

**Comment by ClausKlein** [comment]
## I will not change the include tree, that must be as it is!

### The other rules are less helpful and also partly not usable.

**Comment by wusatosi** [comment]
I think this is more of a discourse question than a exemplar issue?

**Comment by neatudarius** [comment]
> I think this is more of a discourse question than a exemplar issue?

Ofc, any possible change here, will be reflected into the Beman Standard and in all repos. But this thread is long enough to just keep the discussion here (in one place). CC: @wusatosi 

> I will not change the include tree, that must be as it is!

@ClausKlein , I'm not sure what are you talking about. Who proposed a change for `include/`? This issue is about a possible change or preserve of the `examples/` directory? ` David proposed a sync for `examples/` which si not fully regulated in our standard, for consistency.

If we are talking about Beman directories inside the repo, directories such as `include/`, `src/` and `tests/` are regulated by [BEMAN_STANDARD.md#directory-layout](https://github.com/bemanproject/beman/blob/main/docs/BEMAN_STANDARD.md#directory-layout). As stated there, it's a `REQUIREMENT` - I don't see the point in discussion about them in this thread. 

> The other rules are less helpful and also partly not usable.

If you have technical arguments, you can explain them - at any time, the Beman Standard can be updated/changed, if we have reasons to do it (present the problem, the solutions, the reason to pick one, and then we vote, if there is no unanimous consent). We are happy to improve the standard! Feel free to open a discource thread if you plan to start a discussion about directory trees in Beman in general.

Meanwhile, this is the Beman status quo today and until it's changed, all repos MUST respect the Beman Standard. It was discussed and voted by Beman members. No standard breaking changes are allowed in repos (`REQUIREMENT`s rules; some of them are just `RECOMMENDATION`s and can be skipped) - ofc, there are some missing pieces from the document - WIP.

**Comment by neatudarius** [comment]
TLDR:
- current status: `examples/*.cpp`
- proposal from @camio: move examples in `examples/beman/$library/*.cpp`, which can help us when working with multiple Beman libraries in a common parent directory.

@JeffGarland , this is another style topic, like the one from today. Can you please pick a direction here as a style coordinator? I don't think we need to bring this topic into the sync.

**Comment by wusatosi** [comment]
> TLDR:
> 
> * current status: `examples/*.cpp`
> * proposal from [@camio](https://github.com/camio): move examples in `examples/beman/$library/*.cpp`, which can help us when working with multiple Beman libraries in a common parent directory.
> 
> [@JeffGarland](https://github.com/JeffGarland) , this is another style topic, like the one from today. Can you please pick a direction here as a style coordinator? I don't think we need to bring this topic into the sync.

If we agree that we need this directory change, we will need to update the beman standard `[DIRECTORY.EXAMPLES]` and treat this as a universal requirement, we should discuss this during the meeting and move this to discourse.

**Comment by neatudarius** [comment]
> If we agree that we need this directory change, we will need to update the beman standard `[DIRECTORY.EXAMPLES]` and treat this as a universal requirement, 

yes, it's already a requirement. and it's not different from what we require for `src` and `tests`!!!

<img width="1086" alt="Image" src="https://github.com/user-attachments/assets/89846093-0952-4dce-9828-93b935569877" />

> we should discuss this during the meeting and move this to discourse.

I don't think it's valuable to do that. This issue is open since beggining of December. We deferred the style decisions to a person for a reason.

**Comment by wusatosi** [comment]
@neatudarius the style director has spoken, it is up to you on how to move forward. If you think there's no further discussions needed please close this issue.

**Comment by neatudarius** [comment]
> [@neatudarius](https://github.com/neatudarius) the style director has spoken, it is up to you on how to move forward. If you think there's no further discussions needed please close this issue.

I asked Jeff for his intervention. Ofc I agree with him :D. @camio, can I close this issue?

**Comment by camio** [comment]
This isn't something I feel passionate about. I'm happy with the Style Czar's decision here. :-D
